src/App.tsx

```diff
@@ -2,34 +2,37 @@ import React, { useState } from "react";
 import EmailInput from "./components/Auth/EmailInput";
 import Loader from "./components/Shared/Loader";
 import SuccessPage from "./components/Auth/SuccessPage";
-import { authenticateUser } from "./utils/authUtils";
 import logo from './assets/images/dimo-logo.png';
 import "./App.css";
+import { useAuthContext } from "./context/AuthContext";
+import OtpInput from "./components/Auth/OtpInput";
 
 function App() {
+  const { loading } = useAuthContext();  // Get loading state from context
   const [authStep, setAuthStep] = useState(0);  // 0 = Email Input, 1 = Loading, 2 = Success
   const [email, setEmail] = useState("");
 
-  const handleEmailSubmit = (email: string) => {
-    setEmail(email);
-    setAuthStep(1);  // Move to the "Authenticating..." step
-
-    // Simulate authentication
-    authenticateUser(email, () => {
-      setAuthStep(2);  // Move to success page after authentication
-    });
-  };
+  if (loading) {
+    return (
+      <div className="flex h-screen items-center justify-center bg-gray-100">
+        <Loader />
+      </div>
+    );
+  }
 
+  // If not loading, render the rest of the app
   return (
     <div className="flex h-screen items-center justify-center bg-gray-100">
       <div className="bg-white p-8 rounded-lg shadow-lg text-center">
         <img src={logo} alt="Dimo Logo" className="mx-auto mb-6 w-32 h-auto" />
-        {authStep === 0 && <EmailInput onSubmit={handleEmailSubmit} />}
-        {authStep === 1 && <Loader />}
+        
+        {/* Render components based on authStep */}
+        {authStep === 0 && <EmailInput onSubmit={setEmail} setAuthStep={setAuthStep} />}
```

Comment:
Managing steps is still in it's early stages, hence the iterating of steps going on here

Will likely need some pages/routing setup, but will see at what point that becomes necessary